Source/WebInspectorUI/UserInterface/Views/FontVariationDetailsSectionRow.js

```diff
@@ -142,6 +161,92 @@ WI.FontVariationDetailsSectionRow = class FontVariationDetailsSectionRow extends
 
         return 1;
     }
+
+    _showValidity()
+    {
+        if (!this._hasValidValue)
+            this.warningMessage = WI.UIString("Axis value outside of supported range: %s – %s", "A warning that is shown in the Font Details Sidebar when the value for a variation axis is outside of the supported range of values").format(this._formatAxisValueAsString(this._minimumValue), this._formatAxisValueAsString(this._maximumValue));
+        else
+            this.warningMessage = null;
+    }
+
+    _handleValueTextFieldBlur(event)
+    {
+        this._skipUpdatingInputValue = false;
+        this.value = this._value;
+    }
+
+    _handleValueTextFieldInput(event)
+    {
+        let valueAsString = event.target.value;
+        let valueAsNumber = event.target.valueAsNumber;
+
+        if (!/\-?\d+(.\d+)?/.test(valueAsString)) {
+            this._hasValidValue = false;
+            this._skipUpdatingInputValue = true;
```

Comment:
Am I understanding correctly that these two values will always be in sync, unless the user is not actively focused in the text field? If so perhaps `set value` should just ignore attempts to set the value if the field is currently both invalid and focused?

Reply:
That's true. We can leverage the focused state of the input and `this._hasValidValue` in the setter. 
Brilliant suggestion! Love this part of code review: when the same behavior can be achieved with less code :) Thanks! 👍